### PR TITLE
Added context and custom hooks

### PR DESCRIPTION
**Pull request by adars-kafle**

### CODE REVIEW

File: src/context/AuthContext.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// Create the context
+export const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+// Create the Provider Component
+export const AuthProvider = ({ children }: {
+    children: ReactNode
+}) => {
+    const [user, setUser] = useState<User | null>(null);
+
+    const login = async (credentials: LoginCredentials) => {
```
Comment (sudipsth08) [inline src/context/AuthContext.tsx:14]: keep only the data that needs to be share the state across components in context. Login and signup can be handled locally

**Review by sudipsth08 — CHANGES_REQUESTED** [review]

File: src/pages/Suppliers/index.tsx [hunk 2]
```diff
@@ -50,7 +52,14 @@ const SuppliersPage: React.FC = () => {
   };
 
   const handleEditSupplier = (row: MRT_Row<Supplier>) => {
-    console.log("Edit supplier:", row.original);
+    setEditingSupplier(row.original);
```
Comment (sudipsth08) [inline src/pages/Suppliers/index.tsx:55]: Why have you created two states for the same data ?

File: src/pages/Suppliers/index.tsx [hunk 3]
```diff
@@ -215,6 +239,60 @@ const SuppliersPage: React.FC = () => {
           </Button>
         </DialogActions>
       </Dialog>
+
+      {/* Edit Supplier Dialog */}
+      <Dialog open={editOpen} onClose={() => setEditOpen(false)}>
```
Comment (sudipsth08) [inline src/pages/Suppliers/index.tsx:244]: Seperate the dialog into seperate component. Lazy load if possible

**Review by sudipsth08 — CHANGES_REQUESTED** [review]

File: src/context/AuthContext.tsx (hunk 1 above)
Comment (adars-kafle) [reply src/context/AuthContext.tsx]: I removed other functions and only kept the user data in state.

File: src/pages/Suppliers/index.tsx (hunk 2 above)
Comment (adars-kafle) [reply src/pages/Suppliers/index.tsx]: Sorry for the confusion, I have removed the duplicate states for handling supplier information.

File: src/pages/Suppliers/index.tsx (hunk 3 above)
Comment (adars-kafle) [reply src/pages/Suppliers/index.tsx]: I did that as well, thanks for addressing this!

**Pull request merged**